util/src/main/java/io/kubernetes/client/apimachinery/GroupVersionKind.java

```diff
@@ -20,10 +22,7 @@
 
   public GroupVersionKind(String group, String version, String kind) {
     super(group, version);
-    if (kind == null) {
-      throw new IllegalArgumentException("kind must not be null");
-    }
-    this.kind = kind;
+    this.kind = precondition(kind, Objects::isNull, () -> "kind must not be null");
```

Comment:
I don't know how I feel about this from a readability perspective.

@yue9944882 wdyt?

Reply:
@brendanburns interesting. you had no problem with [this, though](https://github.com/kubernetes-client/java/pull/1468/files#diff-f6f5d3fa8282576aad84c7431df956e40e20b563ff1a0f544dc1e9257105e21aR30)

Reply:
i think i improves readability in a way, the refactoring is trivial so it looks good

Reply:
ok. I'm fine with it. The other one feels cleaner somehow, I think it was because it removed two mostly identical assignments, but clearly it's a little inconsistent on my part.